src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -115,10 +115,13 @@ def _register_osrandom_engine(cls):
         # reliably clear the error queue. Once we clear it here we will
         # error on any subsequent unexpected item in the stack.
         cls.lib.ERR_clear_error()
-        cls._osrandom_engine_id = cls.lib.Cryptography_osrandom_engine_id
-        cls._osrandom_engine_name = cls.lib.Cryptography_osrandom_engine_name
-        result = cls.lib.Cryptography_add_osrandom_engine()
-        _openssl_assert(cls.lib, result in (1, 2))
+        if cls.lib.Cryptography_HAS_ENGINE:
+            cls._osrandom_engine_id = cls.lib.Cryptography_osrandom_engine_id
```

Comment:
Why do we do these assignments instead of just having the backend read the attrs from `lib`?

Reply:
Looks like we don't need it any more (if we ever did). We should do this in a separate PR though.